python/cudf/dataframe/multiindex.py

```diff
@@ -0,0 +1,341 @@
+                raise KeyError(element)
+            matches = []
+            for k, code in enumerate(self.codes[self.codes.columns[i]]):
+                if k in validity_mask or len(validity_mask) == 0:
+                    if code == index_of_code_at_level:
+                        matches.append(k)
+            # matches = self.codes.map([j if c[i] == index_of_code_at_level\
+            #       else None for j,c in codes])
+            if len(matches) != 0:
+                validity_mask = matches
+        return validity_mask
+
+    def get_row_major(self, df, row_tuple):
+        valid_indices = self._compute_validity_mask(df, row_tuple)
+        from cudf import Series
+        result = df.take(Series(valid_indices))
+        # Build new index - INDEX based MultiIndex
+        # ---------------
+        from cudf import DataFrame
+        out_index = DataFrame()
+        # Select the last n-k columns where n is the number of source
+        # levels and k is the length of the indexing tuple
+        for k in range(len(row_tuple), len(df.index.levels)):
+            out_index.add_column(df.index.names[k],
+                                 df.index.codes[df.index.codes.columns[k]])
+        # If there's only one column remaining in the output index, convert
+        # it into a StringIndex and name the final index values according
+        # to the proper codes.
+        if len(out_index.columns) == 1:
+            out_index = []
+            for val in result.index.codes[result.index.codes.columns[len(result.index.codes.columns)-1]]:  # noqa: E501
+                out_index.append(result.index.levels[
+                        len(result.index.codes.columns)-1][val])
+            # TODO: Warning! The final index column could be arbitrarily
+            # ordered integers, not Strings, so we need to check for that
+            # dtype and produce a GenericIndex instead of a StringIndex
+            out_index = StringIndex(out_index)
+            out_index.name = result.index.names[len(result.index.names)-1]
+            result.index = out_index
+        else:
+            # Otherwise pop the leftmost levels, names, and codes from the
+            # source index until it has the correct number of columns (n-k)
+            if(len(out_index.columns)) > 0:
+                result.reset_index(drop=True)
+                result.index = result.index._popn(len(row_tuple))
+        # Finally, if n-k==1 and only 1 row is returned, convert the
+        # resulting DataFrame into a Series instead.
+        """
+        print('result')
+        print(len(result))
+        print('n-k')
+        print(len(row_tuple)-len(self.levels))
+        if len(result) == 1 and len(self.levels)-len(row_tuple) == 1:
+            from cudf.dataframe import Series
+            result = Series(result.iloc[0])
+            result.name = row_tuple
+        """
+        return result
+
+    def get_column_major(self, df, row_tuple):
```

Comment:
Similar to above, shouldn't this be a non-public method?

Reply:
Same as above, they are never called by MultiIndex